core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -55,9 +58,9 @@ impl TransactionStateContainer {
         self.priority_queue.is_empty()
     }
 
-    /// Returns the remaining capacity of the queue
-    pub(crate) fn remaining_queue_capacity(&self) -> usize {
-        self.priority_queue.capacity() - self.priority_queue.len()
+    /// Returns the remaining capacity of the container
+    pub(crate) fn remaining_capacity(&self) -> usize {
+        self.priority_queue.capacity() - self.id_to_transaction_state.len()
```

Comment:
This introduces a behavioral change.

Previously we only checked queue length, and when things have been scheduled they are not sitting in the queue.
So it was actually allocating the map beyond it's intended capacity.

With this change, we are constrained to our original (intended) capacity.

Reply:
I wonder what % of the queue is in flight at any given time..